common/src/main/java/io/pravega/common/security/JwtUtils.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.security;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Base64;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Utility methods for JSON Web Tokens (JWT).
+ */
+@Slf4j
+public class JwtUtils {
+
+    /**
+     * The regex pattern for extracting "exp" field from the JWT.
+     *
+     * Examples:
+     *    Input:- {"sub":"subject","aud":"segmentstore","iat":1569837384,"exp":1569837434}, output:- "exp":1569837434
+     *    Input:- {"sub": "subject","aud": "segmentstore","iat": 1569837384,"exp": 1569837434}, output:- "exp": 1569837434
+     */
+    private static final Pattern JWT_EXPIRATION_PATTERN = Pattern.compile("\"exp\":\\s?(\\d+)");
+
+    /**
+     * Extracts expiration time from the specified JWT token.
+     *
+     * @param jsonWebToken the JWT to extract the expiration time from
+     * @return the the expiration time (in seconds). Returns null if
+     *         a) the specified token is blank, or
+     *         b) the specified token is of invalid format, or
+     *         c) expiration time is missing from the token, or
+     *         d) expiration time is not a number.
+     */
+    public static Long extractExpirationTime(String jsonWebToken) {
+        if (jsonWebToken == null || jsonWebToken.trim().equals("")) {
+            return null;
+        }
+        String[] tokenParts = jsonWebToken.split("\\.");
```

Comment:
This is an expensive string parse. Can you just look up `exp` and get the number that succeeds it?

Reply:
It is a pre-existing pice code that was refactored into its own class, and therefore unrelated to the issue at hand. I can raise a separate issue, for this potential performance improvement.

Reply:
Actually, we do look up the expiry string (for example: `"exp": 1569837434`) in another method `parseExpirationTime()`, after first extracting the JWT body and decoding it here.

Reply:
This PR now doesn't use this method anymore. It continues to be used elsewhere.